Classes/Service/AuthenticationService.php

```diff
@@ -317,12 +321,12 @@ protected function convertResourceOwner(array $info)
                     GeneralUtility::intExplode(',', $this->config['usersStoragePid']),
                     Connection::PARAM_INT_ARRAY
                 )),
-                $queryBuilder->expr()->orX(
+                $queryBuilder->expr()->or(
                     $queryBuilder->expr()->eq('tx_oidc', $queryBuilder->createNamedParameter($info['sub'])),
-                    $queryBuilder->expr()->eq('username', $queryBuilder->createNamedParameter($info['email']))
+                    $queryBuilder->expr()->eq('username', $queryBuilder->createNamedParameter($info['email'] ?? ''))
```

Comment:
Not sure if it matters,  but in case we do not get an email address we should not even add this SQL query part. (matching against an empty username is pointless, but you never know what people do in their instances)

Reply:
if we do not consider the e-mail, do we allow reenabling users with another e-mail address?
i think this change is just the status quo with php 8

Reply:
I meant, we should add this condition only if email is not empty.

Reply:
we leave this unchanged for now, done.